sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsPartitionManagement.java

```diff
@@ -80,9 +79,14 @@ void createPartition(
      * @return true if the partition exists, false otherwise
      */
     default boolean partitionExists(InternalRow ident) {
-        String[] partitionNames = partitionSchema().names();
-        String[] requiredNames = Arrays.copyOfRange(partitionNames, 0, ident.numFields());
-        return listPartitionIdentifiers(requiredNames, ident).length > 0;
+      String[] partitionNames = partitionSchema().names();
+      if (ident.numFields() == partitionNames.length) {
+        return listPartitionIdentifiers(partitionNames, ident).length > 0;
+      } else {
+        throw new IllegalArgumentException("The number of fields (" + ident.numFields() +
```

Comment:
@MaxGekk so spark will never call this method with partial partition spec?

Reply:
Currently, it shouldn't especially after this PR https://github.com/apache/spark/pull/30624. Maybe I missed some places but as far as I can see, all calls of `partitionExists()` pass fully specified partition ids.

Reply:
OK, then requiring implementations to fail for this case makes sense to me.

Reply:
As an alternative, we could replace the exact check by an assert. Not sure that it is appropriate solution for default API implementation.